homeassistant/components/elkm1/sensor.py

```diff
@@ -46,6 +62,16 @@ def state(self):
         """Return the state of the sensor."""
         return self._state
 
+    async def async_zone_bypass(self, code=None):
+        """Bypass zone."""
+        if isinstance(self, ElkZone):
```

Comment:
Maybe 
`raise HomeAssistantError("....not an elk zone......")` if its not an ElkZone?

Reply:
Happy to... however, when you try to call a service with an entity that is not registered for that service it silently fails. I didn't raise because that seemed consistent.

For example, calling `elkm1.sensor_zone_bypass` with say an alarm panel "fails" silently (service is not called). Using that logic it seemed right that calling the service with say a `counter` which is also a sensor should fail silently. It just fails later in the processing.

Reply:
Ideally we do not want anything to fail silently as that leaves the user wondering what happens later when they find out their action didn't work. 

It might be a nice improvement to entity services provide user feedback when a service is called with specific entities and we get to https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/service.py#L457 .. but that is rabbit hole we probably do not want to go down since we have to figure out how to handle the some entities fail and some are successful state.